src/metabase/api/user.clj

```diff
@@ -68,13 +68,15 @@
         last_name       (or last_name prev_last_name)]
     (when (or (not= first_name prev_first_name)
               (not= last_name prev_last_name))
-      [first_name last_name])))
+      [(when-not (= first_name :explicitly-nil) first_name)
+       (when-not (= last_name :explicitly-nil) last_name)])))
 
 (defn- maybe-update-user-personal-collection-name! [user-before-update first_name last_name]
   ;; If the user name is updated, we shall also update the personal collection name (if such collection exists).
-  (when-some [[first_name last_name] (updated-user-name user-before-update first_name last_name)]
+  (when-let [[first_name last_name] (updated-user-name user-before-update first_name last_name)]
```

Comment:
Changed this to `when-let` for the following scenario:

Both first/last names are `:explicitly-nil`, therefore `updated-user-name` will return `[nil nil]`, which is still `true` in a when-let but `false` in a `when-some`. We want this true because names are now allowed to be nil.

Note that if first/last names are 'nil' (NOT explicitly set by some API user), then `updated-user-name` returns `nil`, which _is_ false in a `when-let`, so we still get a no-op in that case, which is what we want.

Reply:
This is not how `when-let` and `when-some` differ. `when-some` accepts everything that's not `nil`, `when-let` accepts only truthy values. So the difference is in handling `false`. Both accept `[nil nil]`. (The word some is meant to rhyme with the function `some?` and not with the function `some`.)

Reply:
Hmm.. guess I got confused here. I'm playing in the REPL to improve my mental model of this. Thanks for clarifying

Reply:
`updated-user-name` seems to return `nil` when none of the names have changed.